City-of-Helsinki/drupal-helfi-kymp
# UHF-7251: Fix list of plans cache

# [UHF-7251](https://helsinkisolutionoffice.atlassian.net/browse/UHF-7251)

## What was done
* Invalidates list of plans cache periodically.

## How to install

* Make sure your instance is up and running on correct branch.
  * `git pull origin UHF-7251-fix-list-of-plans-cache`
  * `make fresh`
* Run `make shell` and then:
  * `chmod u+x docker/openshift/crons/invalidate-tags-kymp.sh`
* Turn on caching for your development setup.
* Run `make drush-cr`

## How to test

* Make sure you have a page that contains the list of plans section.
* For easier testing, you can make a temporary local copy of the RSS xml file.
  * Copy the file and place it under the project's `public` directory.
  * Temporary edit the `$rss_feed_url` variable at `public/modules/custom/helfi_kymp_plans/src/ListOfPlans.php` to point to the localhost, e.g. `'http://localhost:8080/temp.xml'`.
* Check that the contents of the list of plans section matches the xml file.
  * Test the site using:
    * a non-Varnish URL for logged-in user.
    * a non-Varnish URL for anonymous user.
    * a Varnish URL (`https://varnish-helfi-kymp.docker.so/`) for anonymous user.
* Make changes to the xml file.
* At this point, the changes shouldn't be visible.
* Run `make shell` and then:
  * `./docker/openshift/crons/invalidate-tags-kymp.sh`
    * (Wait for e.g. 30 seconds and then escape from the command.)
* At this point, the changes should be visible using the non-Varnish URL.
* Run `drush p:queue-work --no-interaction --finish`.
* At this point, the changes should be visible also using the Varnish URL.

## Designers review
